exchange/bidder.go

```diff
@@ -119,6 +123,7 @@ func (bidder *bidderAdapter) requestBid(ctx context.Context, request *openrtb.Bi
 			bidResponse, moreErrs := bidder.Bidder.MakeBids(request, httpInfo.request, httpInfo.response)
 			errs = append(errs, moreErrs...)
 			if bidResponse != nil {
+				seatBid.currency = bidResponse.Currency
```

Comment:
ah, sorry. No. We deliberately chose not to support per-Bid currencies in the ticket, and I think that was the right call.

This only affects Bidders who make several HTTP calls for each `BidRequest`. Although the types on `Bidder` make sure that each HTTP call has the same currency, Prebid can't make sure that each HTTP call returns bids with the same currency as the others.

The question here is: "if a Bidder makes two HTTP calls, and their servers respond with different currencies, what do we do?"

Mine answer would be: first try to convert them using the currency conversion rates (once they're implemented), and then ignore the ones we can't convert and return an error to help publishers debug. Since conversion rates don't exist yet, the best we can do right now is return the errors if they don't match. Let me know if you have any better ideas though...

Reply:
Ok, so, if we don't take into consideration the exchange rate mechanism for the time being, what you are proposing is to check bid response currency against bid request allowed currencies. 
The first approach was to assume that all HTTP calls will end up having the same currency but it's not the case.
I am not sure to understand how it would happen though, how would bidders set differents currencies accross several HTTP calls ? Especially, why wouldn't it be rejected already with the current implementation (sorry but I really want to make sure I fully understand the technical details here :)). Thanks

Reply:
Suppose a publisher makes a BidRequest like:

```
{
  "cur": ["USD", "EUR"],
  "imp": [
    "banner": {
      ...
    },
    "video": {
      ...
    }
  ]
}
```

------------------------------------------------------

The Bidder's servers don't support multiformat, so they split it into two separate HTTP requests:

```
{
  "cur": ["USD", "EUR"],
  "imp": [
    "banner": {
      ...
    },
  ]
}
```

```
{
  "cur": ["USD", "EUR"],
  "imp": [
    "video": {
      ...
    }
  ]
}
```

-------------------------------------------------

According to OpenRTB, either currency is allowed. Based on their inventory, their servers return these two responses:

```
{
  "cur": "USD",
  "seatbid": { ... some Banner bids nested in here ... }
}
```

```
{
  "cur": "EUR",
  "seatbid": { ... some Video bids nested in here ... }
}
```

------------------------------------------------------------------

The adapter's `MakeRequests` function gets called twice--once with each `ResponseData`. One call returns bids in `USD`, and the second returns bids in `EUR`.

This code is responsible for merging the bids from the two calls. In pseudocode, yours does:

```
seatBid = {
  currency: "USD",
  bids: []
}

for each response {
  seatBid.currency = response.currency
  seatBid.bids = append(seatBid.bids, response.bids)
}

return seatBid
```

But... this isn't really true. The Bidder will think they made some bids in `EUR`, and others in `USD`. As written, PBS will report all those bids as being `EUR`, without doing any conversions for them.

Does that make sense? If not, could you point me to the place where you think this is being rejected?

Reply:
Oh, I see, ok it does make sense to me !
I will update the code accordingly then. 

Thanks